crates/red_knot_python_semantic/src/types.rs

```diff
@@ -2111,4 +2110,34 @@ mod tests {
 
         assert_eq!(ty.into_type(&db).repr(&db), expected.into_type(&db));
     }
+
+    #[test]
+    fn type_intersect() {
```

Comment:
This tests more or less the same thing like the Markdown-based test, only in a much more direct way. Let me know if I should remove it.